src/core/binding.c

```diff
@@ -549,6 +549,13 @@ QuicBindingAcceptConnection(
     //
     QuicListenerAcceptConnection(Listener, Connection, Info);
 
+    //
+    // Free the ALPN buffer if it's changed by the app with NewNegotiatedAlpn.
+    //
+    if (NegotiatedAlpn != Connection->Crypto.TlsState.NegotiatedAlpn) {
+        CXPLAT_FREE(NegotiatedAlpn, QUIC_POOL_ALPN);
+    }
```

Comment:
1. Let's move this logic into `QuicListenerClaimConnection`.
2. Do not free the memory if `Connection->Crypto.TlsState.NegotiatedAlpn == Connection->Crypto.TlsState.SmallAlpnBuffer`

Reply:
Done